packages/zoe/test/unitTests/contracts/test-callSpread.js

```diff
@@ -0,0 +1,664 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+import { E } from '@agoric/eventual-send';
+import '../../../exported';
+import { makePromiseKit } from '@agoric/promise-kit';
+import buildManualTimer from '../../../tools/manualTimer';
+
+import { setup } from '../setupBasicMints';
+import { installationPFromSource } from '../installFromSource';
+import { assertPayoutDeposit, assertPayoutAmount } from '../../zoeTestHelpers';
+
+const callSpread = `${__dirname}/../../../src/contracts/callSpread`;
+const simpleExchange = `${__dirname}/../../../src/contracts/simpleExchange`;
+
+function makeFakePriceAuthority(
+  timer,
+  underlyingAmountMath,
+  strikeAmountMath,
+  priceSchedule,
+) {
+  function priceFromSchedule(strikeTime) {
+    let freshestPrice = 0;
+    let freshestTime = -1;
+    for (const tick of priceSchedule) {
+      if (tick.time > freshestTime && tick.time <= strikeTime) {
+        freshestTime = tick.time;
+        freshestPrice = tick.price;
+      }
+    }
+    return freshestPrice;
+  }
+
+  const priceAuthority = {
+    getCurrentPrice: underlyingAmount => {
+      const underlyingValue = underlyingAmountMath.getValue(underlyingAmount);
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now => {
+          const price = priceFromSchedule(now);
+          return strikeAmountMath.make(price * underlyingValue);
+        });
+    },
+    priceAtTime: (timeStamp, underlyingAmount) => {
+      const { promise, resolve } = makePromiseKit();
+
+      underlyingAmountMath.getValue(underlyingAmount);
+      E(timer).setWakeup(
+        timeStamp,
+        harden({
+          wake: () => {
+            return resolve(priceAuthority.getCurrentPrice(underlyingAmount));
+          },
+        }),
+      );
+      return promise;
+    },
+  };
+  return priceAuthority;
+}
+
+// Underlying is in Simoleans. Collateral, strikePrice and Payout are in bucks.
+// Value is in Moola. The price oracle takes an amount in Underlying, and
+// gives the value in Moola.
+test('callSpread below Strike1', async t => {
```

Comment:
It'd be nice to have unit tests of more of the internals of the contract, like the collateralShare calculations.

Reply:
added